handler/instrumented_enqueue_object_test.go

```diff
@@ -0,0 +1,149 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllertest"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"k8s.io/client-go/util/workqueue"
+)
+
+var _ = Describe("InstrumentedEnqueueRequestForObject", func() {
+	var q workqueue.RateLimitingInterface
+	var instance InstrumentedEnqueueRequestForObject
+	var pod *corev1.Pod
+
+	BeforeEach(func() {
+		q = controllertest.Queue{Interface: workqueue.New()}
+		instance = InstrumentedEnqueueRequestForObject{}
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "biz",
+				Name:      "biz",
+			},
+		}
+	})
+	Describe("Create", func() {
+		It("should enqueue a request & emit a metric on a CreateEvent", func() {
+			evt := event.CreateEvent{
+				Object: pod,
+				Meta:   pod.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(1))
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: pod.Namespace,
+					Name:      pod.Name,
+				},
+			}))
+
+			gauges, err := metrics.Registry.Gather()
+			Expect(err).Should(BeNil())
+			Expect(len(gauges)).To(Equal(1))
+			for _, l := range gauges[0].Metric[0].Label {
+				if l.Name == ptrString("name") || l.Name == ptrString("namespace") {
+					Expect(l.Value).To(Equal("biz"))
+				}
+			}
+		})
+	})
+
+	Describe("Delete", func() {
+		It("should enqueue a request & remove the metric on a DeleteEvent", func() {
+			evt := event.DeleteEvent{
+				Object: pod,
+				Meta:   pod.GetObjectMeta(),
+			}
+
+			instance.Delete(evt, q)
+			Expect(q.Len()).To(Equal(1))
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: pod.Namespace,
+					Name:      pod.Name,
+				},
+			}))
+
+			gauges, err := metrics.Registry.Gather()
+			Expect(err).Should(BeNil())
+			Expect(len(gauges)).To(Equal(0))
+		})
+	})
+
+	Describe("Update", func() {
+		It("should enqueue a request in case of UpdateEvent", func() {
+			newpod := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "baz",
+					Name:      "baz",
+				},
+			}
+			evt := event.UpdateEvent{
+				ObjectOld: pod,
+				MetaOld:   pod.GetObjectMeta(),
+				ObjectNew: newpod,
+				MetaNew:   newpod.GetObjectMeta(),
+			}
+
+			instance.Update(evt, q)
+			Expect(q.Len()).To(Equal(2))
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: pod.Namespace,
+					Name:      pod.Name,
+				},
+			}))
+
+			gauges, err := metrics.Registry.Gather()
+			Expect(err).Should(BeNil())
+			Expect(len(gauges)).To(Equal(1))
+			for _, l := range gauges[0].Metric[0].Label {
+				if l.Name == ptrString("name") || l.Name == ptrString("namespace") {
+					Expect(l.Value).To(Equal("biz"))
+				}
+			}
```

Comment:
Maybe write a helper function for this and also check the metric's GVK labels and the metric value?